components/eamxx/src/physics/mam/eamxx_mam_optics_process_interface.cpp

```diff
@@ -0,0 +1,107 @@
+#include <physics/mam/eamxx_mam_optics_process_interface.hpp>
+#include <share/property_checks/field_lower_bound_check.hpp>
+#include <share/property_checks/field_within_interval_check.hpp>
+
+#include "scream_config.h" // for SCREAM_CIME_BUILD
+
+#include <ekat/ekat_assert.hpp>
+
+namespace scream
+{
+
+MAMOptics::MAMOptics(
+    const ekat::Comm& comm,
+    const ekat::ParameterList& params)
+  : AtmosphereProcess(comm, params),
+    aero_config_() {
+}
+
+AtmosphereProcessType MAMOptics::type() const {
+  return AtmosphereProcessType::Physics;
+}
+
+std::string MAMOptics::name() const {
+  return "mam4_optics";
+}
+
+void MAMOptics::set_grids(const std::shared_ptr<const GridsManager> grids_manager) {
+  using namespace ekat::units;
+
+  grid_ = grids_manager->get_grid("Physics");
+  const auto& grid_name = grid_->name();
+
+  ncol_ = grid_->get_num_local_dofs();      // number of columns on this rank
+  nlev_ = grid_->get_num_vertical_levels(); // number of levels per column
+  nswbands_ = 14;                           // number of shortwave bands
```

Comment:
We may want to have a check somewhere that confirms that the number of bands in MAM and in RRTMGP are consistent.  Not sure how to do this, without having something like `get_num_swbands` or equivalent somewhere in the code.

Reply:
Great point. I don't even think we'll know the number of bands before the real MAM optics code makes it in. But when it does, this is something we'll want. (I think I got this number from SPA.)

Reply:
If, say, MAM uses a field (with `nswbands` in its layout) that RRTMGP computes, inside `initialize_impl`, when the fields have been set in all atm procs, MAM can check that there is a provider for that field, by pinging the FieldTracking metadata.

Reply:
Better: we can have a common physics data structure in `physics/share`, storing constants like `nswbands`. The two classes will then ping that data structure for this information.

Reply:
Yet another thought: perhaps we should make the number of bands a CMake configurable parameter for EAMxx, which could end up in our config.h file... @brhillman may have some thought on this.

Reply:
I don't know when the builds were attempted. The timeout may be b/c another (long) job was running (maybe e3sm testing). Adding the retest label should be enough.

Reply:
Ah, wait, we had some issues with mappy over the weekend. I see that all PRs were having issues starting its job. This is likely not an AT issue, but rather a mappy issue. We've been struggling to keep up with its issues... :/